src/Pos/Worker.hs

```diff
@@ -58,6 +60,12 @@ allWorkers NodeResources {..} = mconcatPair
     , wrap' "delegation" $ dlgWorkers
     , wrap' "slotting"   $ (properSlottingWorkers, mempty)
 
+    , case ncTopology ncNetworkConfig of
+        TopologyBehindNAT dnsDomains ->
+          wrap' "subscription" $ subscriptionWorkers ncNetworkConfig dnsDomains
+        _otherwise ->
+          mempty -- No subscription worker required
```

Comment:
Presumably this is where we will insert the kademlia worker for the modes that use it.